packages/next/src/build/webpack-config.ts

```diff
@@ -2483,6 +2433,32 @@ export default async function getBaseWebpackConfig(
       ].filter(Boolean),
     },
     plugins: [
+      isNodeServer &&
+        new webpack.NormalModuleReplacementPlugin(
+          /\.\/(.+)\.shared-runtime$/,
+          function (resource) {
+            const moduleName = path.basename(
+              resource.request,
+              '.shared-runtime'
+            )
+            const layer = resource.contextInfo.issuerLayer
+
+            const runtime =
+              layer === 'app-route-handler'
+                ? 'app-route'
+                : [
+                    WEBPACK_LAYERS.reactServerComponents,
+                    WEBPACK_LAYERS.serverSideRendering,
+                    WEBPACK_LAYERS.appPagesBrowser,
+                    WEBPACK_LAYERS.actionBrowser,
+                    WEBPACK_LAYERS.appRouteHandler,
+                  ].includes(layer as any)
```

Comment:
minor perf nit, extract this to an external variable so it doesn't keep allocating for every resolution

Reply:
I'd actually rewrite as a switch if it were me